jetbrains-core/src/software/aws/toolkits/jetbrains/services/ecs/exec/EcsExecUtils.kt

```diff
@@ -95,6 +105,78 @@ object EcsExecUtils : CoroutineScope by ApplicationThreadPoolScope("EcsExec") {
         )
     }
 
+    fun getEc2InstanceTaskRoleArn(project: Project, clusterArn: String, ecsClient: EcsClient, task: software.amazon.awssdk.services.ecs.model.Task): String? {
+        try {
+            val iamClient = project.awsClient<IamClient>()
+            val containerInstanceArn = task.containerInstanceArn()
+            val res = ecsClient.describeContainerInstances(
+                DescribeContainerInstancesRequest
+                    .builder()
+                    .cluster(clusterArn)
+                    .containerInstances(containerInstanceArn).build()
+            )
+            val ec2InstanceId = res.containerInstances().first().ec2InstanceId()
+            val instanceProfileArn = project.awsClient<Ec2Client>().describeInstances(
+                DescribeInstancesRequest.builder().instanceIds(ec2InstanceId).build()
+            ).reservations().first().instances().first().iamInstanceProfile().arn() ?: return null
+            val instanceProfileName = instanceProfileArn.substringAfter(":instance-profile/")
+            return iamClient.getInstanceProfile(GetInstanceProfileRequest.builder().instanceProfileName(instanceProfileName).build()).instanceProfile()
+                .roles().first().arn() ?: null
+        } catch (e: Exception) {
+            return null
+        }
+    }
+
+    fun getTaskRoleArn(project: Project, clusterArn: String, taskArn: String): String? {
+        val ecsClient = project.awsClient<EcsClient>()
+        val task = ecsClient.describeTasks(DescribeTasksRequest.builder().tasks(taskArn).cluster(clusterArn).build()).tasks().first()
+        return if (task.overrides().taskRoleArn() != null) {
+            task.overrides().taskRoleArn()
+        } else {
+            val roleArn = project.getResourceNow(EcsResources.describeTaskDefinition(task.taskDefinitionArn())).taskRoleArn()
+            if (roleArn == null) {
+                val launchType = task.launchType()
+                if (launchType == LaunchType.EC2) {
+                    val ec2InstanceTaskRoleArn = getEc2InstanceTaskRoleArn(project, clusterArn, ecsClient, task)
+                    ec2InstanceTaskRoleArn
+                } else {
+                    return null
+                }
+            } else {
+                roleArn
+            }
+        }
+    }
+
+    fun checkRequiredPermissions(project: Project, clusterArn: String, taskArn: String): Boolean {
+        try {
+            val iamClient = project.awsClient<IamClient>()
+            val taskRoleArn = getTaskRoleArn(project, clusterArn, taskArn) ?: return false
+            val permissions = listOf(
+                message("session_manager_create_control_channel_permission"),
```

Comment:
this will never be localized so it can just be a const in the file IMO

Reply:
I'll change it to constant...